src/vm/vm_memory/memory_segments.rs

```diff
@@ -68,16 +68,10 @@ impl MemorySegmentManager {
         }
     }
 
-    ///Calculates the size (number of non-none elements) of each memory segment
-    pub fn compute_effective_sizes(&mut self, memory: &Memory) {
-        if self.segment_used_sizes != None {
-            return;
-        }
```

Comment:
Why was this removed?

Reply:
`Option::get_or_insert_with` will not call the closure if it's already present, but will return a reference instead. It's the same behaviour.

Reply:
Sounds good!